### DISCUSSION TITLE
How to ignore `route()` not available in PHPStorm? (TS2304: Cannot find name 'route'.)

### DISCUSSION THREAD

**Discussion by anburocky3** [tighten/ziggy#511] [Q&A]
Showing two errors:
1. ![image](https://user-images.githubusercontent.com/15264938/152610820-ae758f3f-91d1-4ef1-8391-f39ffbd87227.png)
2. ![image](https://user-images.githubusercontent.com/15264938/152610877-7fcc15a9-cfc8-407d-adff-0beecea694b5.png)

Adding extra information:
1. Yes, I'm using `@routes` blade on the layout page and the routes are generated by the laravel.
2. So it means, i can access the `route()` in any vue page right? If that so, IDE is complaining it can't find it. How to fix it? Any solutions?

**Update:**
- I just installed `ziggy-js` and importing it at every page when i need it. And the IDE error goes off. Is this the good method?

**Comment by bakerkretzmar** [comment]
> So it means, i can access the route() in any vue page right?

Yes, that's right. Not sure about the IDE/eslint stuff, the code itself should work fine as far as I can see.

> I just installed ziggy-js and importing it at every page when i need it. And the IDE error goes off. Is this the good method?

No, that's not necessary.

**Reply by anburocky3** [reply]
> Yes, that's right. Not sure about the IDE/eslint stuff, the code itself should work fine as far as I can see.
> 
> >

Yes, by functionality, its working as intended. But since, I'm using PHPStorm and Vue (Typescript). The IDE shows this as an error. Is there any way to avoid this IDE error?

When i installed `ziggy-js`, since I'm importing `route` from that package, the error goes. 

Hope you understand, what i tried to say.

**Reply by bakerkretzmar** [reply]
Yeah I understand, I'm not sure how to turn off that message, sorry 😕

**Reply by pReya** [reply]
This is a Typescript error, not only a eslint error. If we use the global `route()` helper and don't import from `ziggy-js` (which is unnecessary) then obviously Typescript will complain, because it doesn't know the global `route()` helper. There must be a way to properly type the global `route()` in TS, right? Has no one ever done that?

**Reply by anburocky3** [reply]
The only temporary hack is to, import like this

```js
import route from 'ziggy-js';
```
When i need it. But as you said, this is unnecessary, but to avoid IDE errors, we need to do that right now.


If you find any alternative better solutions, please ping.

**Reply by pReya** [reply]
Yeah, that's not a nice way to do it. Ziggy should be declared globally in Typescript, too. I'm still looking into this. Hope I can come up with a solution, soon.

**Reply by talelmishali** [reply]
@pReya  Any luck finding a solution for that?

**Comment by pabueco** [comment]
This should fix it: https://github.com/tighten/ziggy/discussions/516#discussioncomment-3486900
